freelist.go

```diff
@@ -171,7 +174,11 @@ func (f *freelist) read(p *page) {
 	copy(f.ids, ids)
 
 	// Make sure they're sorted.
-	sort.Sort(pgids(f.ids))
+	for i, v := range f.ids[:len(f.ids)-1] {
```

Comment:
Won't [`sort.IsSorted`](https://golang.org/pkg/sort/#IsSorted) perform same amount of iterations as this? I think the advantage there is it will allocate no extra copies for each `pgid` that's iterated upon, unlike this implementation.